src/sentry/static/sentry/app/stores/globalSelectionStore.jsx

```diff
@@ -1,41 +1,78 @@
+import {isEqual} from 'lodash';
 import Reflux from 'reflux';
 
-import {DEFAULT_STATS_PERIOD} from 'app/constants';
-
+import {DATE_TIME} from 'app/components/organizations/globalSelectionHeader/constants';
+import {DEFAULT_STATS_PERIOD, DEFAULT_USE_UTC} from 'app/constants';
+import {isEqualWithDates} from 'app/utils/isEqualWithDates';
 import GlobalSelectionActions from 'app/actions/globalSelectionActions';
 
+const DEFAULT_SELECTION = {
+  projects: [],
+  environments: [],
+  datetime: {
+    [DATE_TIME.START]: null,
+    [DATE_TIME.END]: null,
+    [DATE_TIME.PERIOD]: DEFAULT_STATS_PERIOD,
+    [DATE_TIME.UTC]: DEFAULT_USE_UTC,
+  },
+};
+
 /**
  * Store for global selections
  * Currently stores active project ids for Discover and EventStream
  */
 const GlobalSelectionStore = Reflux.createStore({
   init() {
-    this.selection = {
-      projects: [],
-      environments: [],
-      datetime: {start: null, end: null, range: DEFAULT_STATS_PERIOD},
-    };
+    this.reset();
     this.listenTo(GlobalSelectionActions.updateProjects, this.updateProjects);
     this.listenTo(GlobalSelectionActions.updateDateTime, this.updateDateTime);
     this.listenTo(GlobalSelectionActions.updateEnvironments, this.updateEnvironments);
   },
 
+  reset(state) {
+    this.selection = state || DEFAULT_SELECTION;
+  },
+
   get() {
     return this.selection;
   },
 
   updateProjects(projects = []) {
-    this.selection.projects = projects;
+    if (isEqual(this.selection.projects, projects)) {
+      return;
+    }
+
+    this.selection = {
+      ...this.selection,
+      projects,
+    };
```

Comment:
Why change this from the original `this.selection.projects = projects`? I don't think there's a reason to reassign all of `this.selection`?

Reply:
👍

Reply:
Hmm, I thought calling `this.trigger` takes care of this though... are you saying it doesn't?

Reply:
Ah alright